t/fits.t

```diff
@@ -2,340 +2,480 @@ use strict;
-
-   skip "Astro::FITS::Header not installed", 79 if $@;
+subtest 'header check' => sub {
+
+    my $t = long xvals( zeroes( 11, 20 ) ) - 5;
+    my ( $fh, $file ) = tfile;
+    wfits( $t, $file );    # without a header
+    my $t2 = rfits $file;
+    unlike $t2->hdr->{COMMENT} // '', qr/HASH/,
+      'no "HASH" garbage in written header';
+
+    # note: keywords are converted to uppercase
+    my %hdr =
+      ( 'Foo' => 'foo', 'Bar' => 42, 'NUM' => '0123', NUMSTR => ['0123'] );
+    $t->sethdr( \%hdr );
+    wfits( $t, $file );
+    $t2 = rfits $file;
+    is_pdl $t2, $t, 'w/rfits round-trip';
+    my $h = $t2->gethdr;
+    subtest 'header items' => sub {
+        is $$h{FOO},       "foo",  "FOO";
+        is $$h{BAR},       42,     "BAR";
+        is $$h{'NUM'} + 1, 124,    "NUM";
+        is $$h{'NUMSTR'},  '0123', "NUMSTR";
+    };
+};
 
 ########### Rudimentary table tests ################
+subtest 'Astro::FITS::Header' => sub {
+  SKIP: {
+
+        skip "Astro::FITS::Header not installed", 79
+          if !$PDL::Astro_FITS_Header;
+
+        # note:
+        #   the tests do not directly test the output file,
+        #   instead they write out a file, read it back in, and
+        #   compare to the data used to create the file.
+        #   So it is more of a "self consistent" test.
+
+        subtest 'auto column names, auto column types' => sub {
+
+            my $x      = long( 1, 4, 9, 32 );
+            my $y      = double( 2.3, 4.3, -999.0, 42 );
+            my $table2 = do {
+                my $table = { COLA => $x, COLB => $y };
+                my ( $fh, $file ) = tfile;
+                wfits $table, $file;
+                rfits $file;
+            };
+
+            ok( defined $table2, "Read of table returned something" );
+            is( ref($table2),  "HASH",   "which is a hash reference" );
+            is( $$table2{tbl}, "binary", "and appears to be a binary TABLE" );
+
+            ok( exists $$table2{COLA} && exists $$table2{COLB},
+                "columns COLA and COLB exist" );
+            is( $$table2{hdr}{TTYPE1}, "COLA", "column #1 is COLA" );
+            is( $$table2{hdr}{TFORM1}, "1J",   "  stored as 1J" );
+            is( $$table2{hdr}{TTYPE2}, "COLB", "column #2 is COLB" );
+            is( $$table2{hdr}{TFORM2}, "1D",   "  stored as 1D" );
+
+            is_pdl $x, $$table2{COLA}, "COLA";
+            is_pdl $y, $$table2{COLB}, "COLB";
+
+        };
+
+        subtest 'explicit column name, auto column type' => sub {
+
+            my $x = long( 1, 4, 9, 32 );
+            my $y = double( 2.3, 4.3, -999.0, 42 );
+
+            my $table2 = do {
+                my $table = {
+                    BAR => $x,
+                    FOO => $y,
+                    hdr => { TTYPE1 => 'FOO', TTYPE2 => 'BAR' }
+                };
+                my ( $fh, $file ) = tfile;
+                wfits $table, $file;
+                rfits $file;
+            };
+
+            ok(
```

Comment:
Please do here the multiple checks you did above. A single "not OK" would be a rather useless test diagnostic.